tests/FSharp.Test.Utilities/ProjectGeneration.fs

```diff
@@ -442,6 +451,37 @@ type ProjectWorkflowBuilder(initialProject: SyntheticProject, ?checker: FSharpCh
             return { ctx with Signatures = ctx.Signatures.Add(fileId, newSignature) }
         }
 
+    /// Find a symbol using the provided range, mimicing placing a cursor on it in IDE scenarios
+    [<CustomOperation "placeCursor">]
+    member this.PlaceCursor(workflow: Async<WorkflowContext>, fileId, line, colAtEndOfNames, fullLine, symbolNames) = 
+        async {
+            let! ctx = workflow
+            let! results = checkFile fileId ctx.Project checker
+            let typeCheckResults = getTypeCheckResult results
+
+            let su = typeCheckResults.GetSymbolUseAtLocation(line,colAtEndOfNames,fullLine,symbolNames)
+
+            return {ctx with Cursor = su}
+        }
+
+
+
+    /// Find all references within a single file, results are provided to the 'processResults' function
+    [<CustomOperation "findAllReferences">]
+    member this.FindAllReferences(workflow: Async<WorkflowContext>, fileId: string,  processResults) =
+        async{
+            let! ctx = workflow
+            let po = ctx.Project.GetProjectOptions checker          
+            let s = ctx.Cursor |> Option.defaultWith (fun () -> failwith "Please place cursor at a valid location via placeCursor first")
+            let file = ctx.Project.Find fileId           
+            let absFileName = ctx.Project.ProjectDir ++ file.FileName
+            let! results = checker.FindBackgroundReferencesInFile(absFileName,po, s.Symbol)
```

Comment:
Small question for my education: do you know why it is called "background references"? Like if this is just references or if there is any special meaning.

Reply:
IMO background means that it operates also besides the currently open document.
And the full document text is not input to this action, but existing cached data is used (or read from FileSystem).